packages/tinacms/src/tina-cms/alerts.ts

```diff
@@ -0,0 +1,80 @@
+/**
+
+ Copyright 2019 Forestry.io Inc
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ */
+
+import { Subscribable } from '@tinacms/core'
+
+export class Alerts extends Subscribable {
+  private alerts: Map<string, Alert> = new Map()
+
+  private add(
+    level: AlertLevel,
+    message: string,
+    timeout: number = 3000
+  ): () => void {
+    const alert = {
+      level,
+      message,
+      timeout,
+    }
+
+    this.alerts.set(alert.message, alert)
```

Comment:
This will cause two alerts with the same message to overwrite each other. I'm guessing this is a feature and not a bug, but is there a scenario where we might not want that to happen?

Reply:
Honestly not sure 🤷‍♂

Reply:
because pushing subsequent alerts with the same message doesn't even change their appearance in alert order, in the event 2 alerts with the same message (e.g. "saved successfully") are pushed, there is no visual feedback

therefore I have elected to generate a unique ID per alert